### ISSUE TITLE
[Bug]: Starting server immediately after loading large campaign throws error

### ISSUE THREAD

**Issue by aeauseth** [RPTools/maptool#3756]
### Describe the Bug

Immediately starting the server (before the map has a chance to refresh) will show the "Server has disconnected" error.  Subsequent "You are already running a server" and "Could not restart personal server" errors will show but @thelsing suggested these are follow-up errors.  This is likely only to occur with larger campains as they can be slow (30+ seconds) to load. #3753 is supposed to improve load times.

Despite these errors, the server does seem to start and clients can connect.

![ServerHasDicsonnected](https://user-images.githubusercontent.com/13549766/204370663-637b4883-2d31-403d-9f3e-967011f78cde.png)

![startserverpng](https://user-images.githubusercontent.com/13549766/204372050-afee19b1-655a-4b57-b997-1a3c5376eb62.png)

### To Reproduce

1. Load a large campaign.
2. Immediately start the server before the map has a chance to refresh.

### Expected Behaviour

Should not be able to start a server if not ready to do so.  Consider one or more of the following:

1. Disabling teh start server menu option until server is ready to start without errors.
2. Alert that the server is not ready to be started.  
3. Progress bar or other delay when server start is requested, delaying until ready to actually start the server.

### MapTool Info

Maptool v1.12.2

### Desktop

windows

### Additional Context

[maptool.log](https://github.com/RPTools/maptool/files/10107063/maptool.log)

**Comment by FullBleed** [comment]
If I may add... 

The "please wait for map to refresh" warning after starting a server has always been something of a mystery to me... Once I start a server I'm always having to tell my player, "Hold on... wait... don't connect yet..."

*twiddle thumbs*

"Ok... go ahead and connect."

It would be nice if the process of starting a server didn't need such a warning.  I'd prefer it if it didn't say "Server Started" until it was actually ready/safe to receive connections.

**Comment by adventuremagic123** [comment]
Yes, I think the entire UI should be locked until rhe campaign is loaded.  A message should be constantly displayed somewhere saying "Campaign loading ...".
